src/celeritas/ext/ScopedRootErrorHandler.cc

```diff
@@ -18,6 +18,9 @@ namespace celeritas
 {
 namespace
 {
+//---------------------------------------------------------------------------//
+bool g_has_root_errored_{false};
```

Comment:
Can this be read and/or updated from multiple thread?

Reply:
Can `SetErrorHandler` be called from multiple threads simultaneously? I feel like using this inside a "worker" thread is outside the rational use case...

Reply:
Indeed, `SetErrorHandler` can not either be called from multiple thread (without an explicit lock) and would indeed be semantically ambiguous.